### DISCUSSION TITLE
Creating a pull request encountered 'Base branch does not allow updates. 's error.

### DISCUSSION THREAD

**Discussion by Leaovo-man2** [community/community#65403] [Pull Requests] [answered]
### Select Topic Area

Bug

### Body

I'm planning to contribute code to a repository, and after creating a pull request, I get an error:
![image](https://github.com/community/community/assets/132978937/0bb0592c-1c99-4c52-bf18-cf1abfa489e3)

**Comment by lab596** [comment]
"Base branch does not allow updates" error while creating a pull request  indicates that the repository's settings prevent updates to the base branch. This might be due to branch protection rules, such as requiring pull requests to be up to date with the base branch before they can be merged.

1. Try to Rebase Your Branch:
Ensure that your feature branch is up to date with the latest changes from the base branch. You can do this by rebasing your branch onto the base branch. This action replays your changes on top of the latest base branch changes.

2. Contact Repository Maintainers:
If you've tried to rebase and force-push your commits to that branch and are still encountering the error, it's possible that the repository maintainers have specific rules or settings in place that you're not aware of. Reach out to them for assistance or guidance on how to properly contribute your changes.

3. Follow Repository Guidelines:
Make sure you're following the repository's contribution guidelines and development workflow. Some repositories have specific requirements for creating pull requests and maintaining branches, so familiarize yourself with those guidelines.

**Reply by lazarljubenovic** [reply]
We can ask ChatGPT ourselves, you know.

**Comment by seyLu** [comment, marked as answer]
### In the case of contributing to another repo via forking
- first create a new branch
> creating a branch from `main` branch is like copying the contents of the `main` branch, and making changes to the copy (without affecting the `main` branch)

### you can do this via CLI
  
```bash
# assuming you are in main branch
git checkout -b <name_of_new_branch>
  
# if not, first checkout to main, then create the branch
git checkout main
git checkout -b <name_of_new_branch>
```
  
### or via github
- click on main branch button
- click view all branches
- click new branch
- write the name of the new branch
- make sure the source branch is `main`
- click create new branch
- click the new branch to switch to that branch
  
![image](https://github.com/community/community/assets/98249191/f6a39866-01a4-4633-9ce1-ad34b3c6e44f)
![image](https://github.com/community/community/assets/98249191/7009f52b-6757-4400-9659-54bcd896992f)
![image](https://github.com/community/community/assets/98249191/c8d0db59-1025-444a-85c1-8e29a5788889)
![image](https://github.com/community/community/assets/98249191/ba64d79b-8283-45cd-bf71-0671d93c57b4)

### You make changes to the branch and Commit as usual

### If using CLI when using `git push`, your local computer doesn't know that it should point that new local branch to a new remote branch so you have to explicitly tell it to do so
 
```bash
# this will create and point to a remote branch in github with the same name as the local branch
git push --set-upstream origin <name_of_new_branch>
```

### If using github
- after making changes, click commit changes button
- write a commit message
- make sure you are committing to the <name_of_new_branch> branch
- click commit changes

![image](https://github.com/community/community/assets/98249191/2900b597-8a65-4318-9c1f-c1eedc7cb72b)
![image](https://github.com/community/community/assets/98249191/678fa86f-c2cd-401f-b0c0-b6a6a7cf87bd)


### Once you're done, you create and submit a Pull Request
- if you go to the repo, you will see a banner
- click compare & pull request
- supply a title
- describe the changes you've made
- click create pull request

![image](https://github.com/community/community/assets/98249191/99ee7a12-e168-4692-9bd5-9180d1243a0a)
![image](https://github.com/community/community/assets/98249191/a2629834-e13c-4381-93b3-4c9d3a74964f)

### The original owner(s) or maintainer(s) of the repo will then review your pull request

### If it's up to par with their standards, then and only then you're changes will be merged to `main`

you can read more on creating PRs here, an fcc article: 
https://www.freecodecamp.org/news/how-to-make-your-first-pull-request-on-github-3

**Comment by b-opacic-bm** [comment]
`settings -> branches -> [your_branch] -> edit -> untick the "lock branch"`
Fixed for me, hope it helps somebody 🤸

**Reply by ninebelowzero** [reply]
Thank you, I was the somebody :-)

**Reply by ivan-spichka** [reply]
thanks, this was helpful)

**Reply by frederikheld** [reply]
Is it relevant when this setting was changed? It feels like branches that have been opened when "lock branch" was active on main keep showing the "merging is blocked" warning.

**Comment by noahwelch** [comment]
I just ran into this and it required a different solution than I saw here. My problem was that in addition to branch protection rules at `/settings/branches`, there was also a ruleset with conflicting rules on my base branch. I went to `/settings/rules` and deleted the ruleset.
